Source/JavaScriptCore/runtime/StringPrototypeInlines.h

```diff
@@ -377,4 +377,45 @@ inline JSString* replaceUsingStringSearch(VM& vm, JSGlobalObject* globalObject,
     RELEASE_AND_RETURN(scope, jsSpliceSubstringsWithSeparators(globalObject, jsString, string, sourceRanges.data(), sourceRanges.size(), replacements.data(), replacements.size()));
 }
 
+inline JSString* tryReplaceOneCharUsingString(JSGlobalObject* globalObject, JSString* string, JSString* search, JSString* replacement)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    if (!string->isNonSubstringRope() || string->length() > 0xFF)
```

Comment:
I think Substring rope should be supported too

Reply:
But the issue is that we are currently not allowing substring of substring, so yeah, we need FIXME comment here.

Reply:
And this is opposite. We should do optimization when string is >= 0xff.
So,
`string->length() < 0xff`